src/main/scala/it/pagopa/interop/purposeprocess/api/impl/PurposeApiServiceImpl.scala

```diff
@@ -498,6 +498,42 @@ final case class PurposeApiServiceImpl(
     onComplete(result) { suspendPurposeVersionResponse[PurposeVersion](operationLabel)(suspendPurposeVersion200) }
   }
 
+  override def rejectPurposeVersion(purposeId: String, versionId: String, payload: RejectPurposeVersionPayload)(implicit
+    contexts: Seq[(String, String)],
+    toEntityMarshallerProblem: ToEntityMarshaller[Problem]
+  ): Route =
+    authorize(ADMIN_ROLE) {
+      val operationLabel = s"Rejecting Version $versionId of Purpose $purposeId"
+      logger.info(operationLabel)
+
+      val result: Future[Unit] = for {
+        purposeUUID    <- purposeId.toFutureUUID
+        versionUUID    <- versionId.toFutureUUID
+        organizationId <- getOrganizationIdFutureUUID(contexts)
+        purpose        <- purposeManagementService.getPurposeById(purposeUUID)
+        _              <- assertOrganizationIsAProducer(organizationId, purpose.eserviceId)
+        _              <- getVersion(purpose, versionUUID)
+        _              <- purposeManagementService.rejectPurposeVersion(
+          purposeUUID,
+          versionUUID,
+          PurposeManagementDependency.RejectPurposeVersionPayload(
+            rejectionReason = payload.rejectionReason,
+            stateChangeDetails = PurposeManagementDependency.StateChangeDetails(
+              PurposeManagementDependency.ChangedBy.PRODUCER,
+              dateTimeSupplier.get()
+            )
+          )
+        )
+        _              <- authorizationManagementService.updateStateOnClients(
+          purposeId = purposeUUID,
+          versionId = versionUUID,
+          state = AuthorizationManagementDependency.ClientComponentState.INACTIVE
+        )
```

Comment:
This part is not necessary.
@galales, are you agree?

Reply:
I agree, only a `WaitingForApproval` version can be rejected, and in that state it cannot be in `ACTIVE` state on the client, so disabling it is not necessary